spec/models/user_spec.rb

```diff
@@ -0,0 +1,17 @@
+require_relative '../rails_helper'
+
+RSpec.describe User, type: :model do
+  subject(:user) { described_class.new(name: 'John Doe', posts_counter: 0) }
+
+  describe 'associations' do
+    it { should have_many(:posts).with_foreign_key(:author_id) }
+    it { should have_many(:comments).with_foreign_key(:author_id) }
+    it { should have_many(:likes).with_foreign_key(:author_id) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_length_of(:name).is_at_least(3).is_at_most(50) }
+    it { should validate_numericality_of(:posts_counter).only_integer.is_greater_than_or_equal_to(0) }
+  end
```

Comment:
- [ ] As required for the project any custom methods in the models to have a corresponding unit test, Kindly ensure that you add unit test(s) to `#recent_posts` method in the `User` class.